src/dataflow/components/nodes/factories/data-storage-rete-node-factory.tsx

```diff
@@ -0,0 +1,111 @@
+import Rete from "rete";
+import { Node, Socket } from "rete";
+import { NodeData } from "rete/types/core/data";
+import { DataflowReteNodeFactory } from "./dataflow-rete-node-factory";
+import { NumControl } from "../controls/num-control";
+import { TextControl } from "../controls/text-control";
+import { PlotControl } from "../controls/plot-control";
+import * as uuid from "uuid/v4";
+
+export class DataStorageReteNodeFactory extends DataflowReteNodeFactory {
+  constructor(numSocket: Socket) {
+    super("Data Storage", numSocket);
+  }
+
+  public builder(node: Node) {
+    const inputUuid = uuid();
+    const inp = new Rete.Input("i-" + inputUuid, "sequence", this.numSocket);
+    inp.addControl(new TextControl(this.editor, "it-" + inputUuid, node, "", "my-sequence"));
+
+    return node
+      .addControl(new TextControl(this.editor, "dataset", node, "name", "my-dataset"))
+      .addControl(new NumControl(this.editor, "interval", node, false, "interval", 1, 1))
+      .addControl(new PlotControl(this.editor, "plot", node))
+      .addInput(inp) as any;
+  }
+
+  public worker(node: NodeData, inputs: any, outputs: any) {
+    const inputValues: any = {};
+    const inputKeys: any = Object.keys(inputs);
+    if (inputKeys) {
+      inputKeys.forEach((key: any) => {
+        if (inputs[key].length) {
+          const inputValue = { name: "sequence", val: inputs[key][0] };
+          inputValues[key] = inputValue;
+        }
+      });
+    }
+
+    if (this.editor) {
+      const _node = this.editor.nodes.find((n: { id: any; }) => n.id === node.id);
+      if (_node) {
+        this.updateInputs(_node);
+        _node.data.nodeValue = inputValues;
+        this.editor.view.updateConnections( {node: _node} );
+      }
+    }
+  }
+
+  private updateInputs(node: Node) {
+    this.addInputs(node);
+    this.removeInputs(node);
+  }
+
+  private addInputs(node: Node) {
+    const iterator = node.inputs.values();
+    let result = iterator.next();
+    let lastInput = result.value;
+    while (!result.done) {
+      lastInput = result.value;
+      result = iterator.next();
+    }
```

Comment:
An iterator did the job here.  Maybe an easier way?

Reply:
We run into trouble with this approach though since `node.inputs` is a `Map`.  Calling `Object.keys(node.inputs)` returns an empty array.  We can call `node.inputs.keys` which returns an iterator with the keys instead of a simple array with the keys.

Reply:
Ah, a little investigation suggests that you can get an array of keys with:
```
const keys = [...node.inputs.keys()];
```

Reply:
That throws a typescript error: 
`Type 'IterableIterator<string>' is not an array type or a string type`

Seems like there should be a simple, one line way to get the last element of a map.

Reply:
Aha!  You got me on the right track.  This can get the last input entry in the node.inputs map in 2 lines:
```const lastKey = Array.from( node.inputs.keys() )[node.inputs.size-1];
const lastInput = node.inputs.get(lastKey);```

Reply:
Convert the keys to an array and use the size to get the last entry in the new key array.  Then use `map.get` with the key that we just obtained to get the final entry in the map.

Reply:
I suppose that could be collapsed into a single line, but it's pretty hard to read :)